src/main/java/io/druid/data/input/impl/DimensionsSpec.java

```diff
@@ -72,10 +118,58 @@ public DimensionsSpec(
     return dimensionExclusions;
   }
 
-  @JsonProperty
+  @Deprecated @JsonIgnore
   public List<SpatialDimensionSchema> getSpatialDimensions()
   {
-    return spatialDimensions;
+    //TODO: make this return DimensionSchema list instead
```

Comment:
generally we avoid committing TODOs, this should either be done now or file an issue to do it later.